src/cascadia/QueryExtension/ExtensionPalette.cpp

```diff
@@ -286,12 +288,14 @@ namespace winrt::Microsoft::Terminal::Query::Extension::implementation
         {
             auto suggestion = winrt::to_string(selectedItemAsChatMessage.MessageContent());
 
-            // the AI sometimes sends code blocks with newlines in them
-            // sendInput doesn't work with single new lines, so we replace them with \r
+            // the AI sometimes sends multiline code blocks
+            // we don't want to run any of those commands when the chat item is clicked,
+            // so we replace newlines with the appropriate delimiter
             size_t pos = 0;
             while ((pos = suggestion.find("\n", pos)) != std::string::npos)
             {
-                suggestion.replace(pos, 1, "\r");
+                const auto delimiter = _ActiveCommandline == cmdExe ? cmdCommandDelimiter : commandDelimiter;
```

Comment:
have we already stripped and normalized the path, such that cmd.exe will always match `cmd.exe` no matter how it was specified?

Reply:
Yep, we do this [here](https://github.com/microsoft/terminal/blob/489a0f082d16e0ed5fe40f31eaf08ea7aeaf0838/src/cascadia/TerminalApp/TerminalPage.cpp#L5620)

Reply:
My bad, turns out just `cmd` was _not_ getting mapped to `cmd.exe`, so now we check for that as well